bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResponseQueueReader.java

```diff
@@ -123,7 +126,7 @@ public synchronized FlatRow getNextMergedRow() throws IOException {
   protected ResultQueueEntry<FlatRow> getNext() throws IOException {
     ResultQueueEntry<FlatRow> queueEntry;
     try {
-      queueEntry = resultQueue.take();
+      queueEntry = resultQueue.poll(waitForRowsTimeoutMillis, TimeUnit.MILLISECONDS);
       // Should never happen
       if (queueEntry == null) {
         throw new IllegalStateException("Timed out awaiting next sync rows");
```

Comment:
Please remove the comment now that this is something expected.

Also please improve the exception thrown to the user. I dont think IllegalStateException is appropriate anymore and the message can use a bit of TLC